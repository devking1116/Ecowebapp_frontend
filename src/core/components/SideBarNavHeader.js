/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Icon } from 'semantic-ui-react';

type Props = {
  isExpanded?: bool,
  expandSideBar?: () => void,
  collapseSideBar?: () => void,
};

/**
 * A header that can displayed in expanded or collapsed formate, for side bars used for navigation.
 */
export default function SideBarNavHeader(props: Props): React.Element<*> {
  const { isExpanded, expandSideBar, collapseSideBar, ...otherProps } = props;

  const styles = reactCSS({
    'default': {
      wrapper: {
        display: 'flex',
        justifyContent: 'flex-end',
        height: 20,
      },
    },
  });

  return (
    <div style={styles.wrapper} {...otherProps}>
    {isExpanded?
      <Icon name="minus circle" onClick={collapseSideBar} />:
      <Icon name="content" size="large" onClick={expandSideBar} />
    }
    </div>
  );
}
