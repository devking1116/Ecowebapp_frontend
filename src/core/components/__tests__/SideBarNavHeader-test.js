import React from 'react';
import renderer from 'react-test-renderer';

import SideBarNavHeader from '../SideBarNavHeader';

test('when expanded, it displays it the close icon.', () => {
  const component = renderer.create(
    <SideBarNavHeader isExpanded />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when not expanded, it displays it the open icon.', () => {
  const component = renderer.create(
    <SideBarNavHeader />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when clicked, if expanded, it calls the collapse function.', () => {
  const collapseFn = jest.fn();
  const component = renderer.create(
    <SideBarNavHeader isExpanded collapseSideBar={collapseFn} />
  );
  const tree = component.toJSON();
  const [ icon ] = tree.children;
  const { onClick } = icon.props;

  onClick();
  expect(collapseFn).toHaveBeenCalledTimes(1);
});

test('when clicked, if not expanded, it calls the expand function.', () => {
  const expandFn = jest.fn();
  const component = renderer.create(
    <SideBarNavHeader expandSideBar={expandFn} />
  );
  const tree = component.toJSON();
  const [ icon ] = tree.children;
  const { onClick } = icon.props;

  onClick();
  expect(expandFn).toHaveBeenCalledTimes(1);
});
