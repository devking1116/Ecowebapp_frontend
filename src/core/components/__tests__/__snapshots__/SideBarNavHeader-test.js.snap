exports[`test when expanded, it displays it the close icon. 1`] = `
<div
  style={
    Object {
      "WebkitJustifyContent": "flex-end",
      "display": "flex",
      "height": 20,
      "justifyContent": "flex-end",
    }
  }>
  <i
    aria-hidden="true"
    className="minus circle icon"
    onClick={undefined} />
</div>
`;

exports[`test when not expanded, it displays it the open icon. 1`] = `
<div
  style={
    Object {
      "WebkitJustifyContent": "flex-end",
      "display": "flex",
      "height": 20,
      "justifyContent": "flex-end",
    }
  }>
  <i
    aria-hidden="true"
    className="content large icon"
    onClick={undefined} />
</div>
`;
